shell/platform/android/io/flutter/view/AccessibilityBridge.java

```diff
@@ -824,11 +829,23 @@ public AccessibilityNodeInfo createAccessibilityNodeInfo(int virtualViewId) {
       result.setText(semanticsNode.getValueLabelHint());
     } else if (!semanticsNode.hasFlag(Flag.SCOPES_ROUTE)) {
       CharSequence content = semanticsNode.getValueLabelHint();
+      if (Build.VERSION.SDK_INT < Build.VERSION_CODES.P) {
+        if (semanticsNode.tooltip != null) {
+          content = content != null ? content : "";
+          content = content + "\n" + semanticsNode.tooltip;
+        }
+      }
       if (content != null) {
         result.setContentDescription(content);
       }
     }
 
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
```

Comment:
what is the intended behavior below this version?  The code in L832 required `!semanticsNode.hasFlag(Flag.SCOPES_ROUTE)`, is it always true in API level < Build.VERSION_CODES.P. 

If not, why?

Reply:
SCOPES_ROUTE doesn't need content description because it is not a11y focusable. This is already the case before the change.

Reply:
and unfortunately, I don't know how to set up test to be run in order API level. It complains about missing binary if i set the config in Robolectric test.

Reply:
could we add comments. Sorry, I think this will help the next person trying to contribute :)

Reply:
could we add comments? Sorry, I think this will help the next person trying to contribute :)

Imagine that someone jumps into this code for this first time, and that person is trying to make sense of each case, etc...